server/command.go

```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
+		})
+		counter++
+		if findErr != nil {
+			return 0, responsef("Error updating post: %s", findErr.Message)
+		}
+	}
+	return counter, nil
+}
+
+func (p *Plugin) executeCommandReQueue(args *model.CommandArgs) *model.CommandResponse {
+	split := strings.Fields(args.Command)
+
+	if len(split) <= 2 {
+		return responsef("Missing parameters for requeue command")
+	}
+
+	meeting, err := p.GetMeeting(args.ChannelId)
+	if err != nil {
+		return responsef("Can't find the meeting")
 	}
 
-	postList := *searchResults.PostList
-	numQueueItems := len(postList.Posts)
+	oldPostID := split[2]
+	postToBeReQueued, _ := p.API.GetPost(oldPostID)
+	hashtagDateFormat, parsedMeetingMessage, errors := parseMeetingPost(meeting, postToBeReQueued)
+	if errors != nil {
+		return errors
+	}
+
+	originalPostDate := strings.ReplaceAll(strings.TrimSpace(parsedMeetingMessage.date), "_", " ") // reverse what we do to make it a valid hashtag
+	originalPostMessage := strings.TrimSpace(parsedMeetingMessage.textMessage)
+
+	today := time.Now()
+	local, _ := time.LoadLocation("Local")
+	formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
+	if formattedDate.Year() == 0 {
+		thisYear := today.Year()
+		formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+	}
 
-	_, appErr = p.API.CreatePost(&model.Post{
+	if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+		return responsef("We don't support re-queuing future items, only available for present and past items.")
+	}
+
+	hashtag, error := p.GenerateHashtag(args.ChannelId, false, -1, true, formattedDate.Weekday())
+	if error != nil {
+		return responsef("Error calculating hashtags. Check the meeting settings for this channel.")
+	}
+
+	numQueueItems, itemErr := calculateQueItemNumberAndUpdateOldItems(meeting, args, p, hashtag)
+	if itemErr != nil {
+		return itemErr
+	}
+
+	_, appErr := p.API.UpdatePost(&model.Post{
+		Id:        oldPostID,
 		UserId:    args.UserId,
 		ChannelId: args.ChannelId,
 		RootId:    args.RootId,
-		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems+1, message),
+		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems, originalPostMessage),
 	})
 	if appErr != nil {
-		return responsef("Error creating post: %s", appErr.Message)
+		return responsef("Error updating post: %s", appErr.Message)
 	}
+	return &model.CommandResponse{Text: fmt.Sprintf("Item has been Re-queued to %v", hashtag), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+}
 
-	return &model.CommandResponse{}
+func parseMeetingPost(meeting *Meeting, post *model.Post) (string, ParsedMeetingMessage, *model.CommandResponse) {
+	var (
+		prefix            string
+		hashtagDateFormat string
+	)
+	if matchGroups := meetingDateFormatRegex.FindStringSubmatch(meeting.HashtagFormat); len(matchGroups) == 4 {
+		prefix = matchGroups[1]
+		hashtagDateFormat = strings.TrimSpace(matchGroups[2])
+	} else {
+		return "", ParsedMeetingMessage{}, responsef("Error 267")
```

Comment:
What is `Error 267`?

Reply:
Will make it friendly @maisnamrajusingh @mickmister .